ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Assertions.scala

```diff
@@ -34,14 +35,21 @@ object Assertions extends DiffExtensions {
       )
   }
 
+  /** Match the given exception against a status code and a regex for the expected message.
+      Succeeds if the exception is a GrpcException with the expected code and
+      the regex matches some part of the message or there is no message and the pattern is
+      empty.
+    */
   def assertGrpcError(t: Throwable, expectedCode: Status.Code, pattern: String): Unit =
```

Comment:
Perhaps provide an overload that takes a `Pattern`, rather than making everyone who calls this think about regular expressions?

I'd be up for renaming this one to `substring`.

Reply:
good idea, added an overload. I don’t care about the name so happy to change it instead of overloading if you prefer.

But we should all spend more time thinking about regex, regex are pretty cool!